### PR TITLE
Functionality/fetch request promise

### PR DESCRIPTION
**Pull request by michaelputnam67**
# Description

Add fetch request functionality. 

Addresses # (issue)
#30 
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Dev Tools

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
 // Your fetch requests will live here!
+const userData = fetch('https://what-s-cookin-starter-kit.herokuapp.com/api/v1/users').then(response => response.json())
 
+const recipeData = fetch('https://what-s-cookin-starter-kit.herokuapp.com/api/v1/recipes').then(response => response.json())
 
-// console.log('I will be a fetch request!')
+const ingredientData = fetch('https://what-s-cookin-starter-kit.herokuapp.com/api/v1/ingredients').then(response => response.json())
+
+let apiCalls = Promise.all([userData, recipeData, ingredientData])
```
Comment (laytonmaes) [inline src/apiCalls.js:8]: this was a really good call

File: README.md [hunk 2]
```diff
@@ -14,3 +14,4 @@ The details of this project are outlined in the <a href="https://frontend.turing
 ## Testing
 
 Mocha and chai are already set up, with a boilerplate test for you.
+# iron-fe
```
Comment (angelewilliams) [inline README.md:17]: ?

File: src/scripts.js [hunk 3]
```diff
@@ -10,6 +10,28 @@ import RecipeRepository from '../src/classes/RecipeRepository';
 import User from '../src/classes/User';
 const { usersData } = require('../src/data/users')
 
+
+
+const instantiateClasses = (recipeData, ingredientData, userData) => {
```
Comment (laytonmaes) [inline src/scripts.js:15]: i like how we implemented this. good work!

File: src/classes/Recipe.js [hunk 4]
```diff
@@ -12,14 +10,14 @@ class Recipe {
 		this.tags = recipe.tags;
 		this.saved = false;
 		this.wantToCook = false
-		this.displayedTag;
-		this.ingredients;
-		this.ingredientNames;
+		// this.displayedTag;
+		this.ingredients = this.collectIngredients(ingredientData)
```
Comment (angelewilliams) [inline src/classes/Recipe.js:14]: Glad we figured this one out!

File: src/scripts.js (hunk 3 above)
Comment (angelewilliams) [inline src/scripts.js:15]: Really cool how you figured this out!

File: src/scripts.js [hunk 5]
```diff
@@ -28,70 +50,70 @@ let savedRecipes = document.getElementById('saveRecipes')
...
 //~~~~~~~~~~~~~~~~~~~~ EVENT LISTENERS  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-window.addEventListener('load', () => {
-  initiatePage()
-});
+// window.addEventListener('load', () => {
+//   initiatePage()
+// });
+
+
+const createEventListeners = (recipeRepository, user) => {
```
Comment (angelewilliams) [inline src/scripts.js:61]: This was such a breakthrough!

**Review by angelewilliams — APPROVED** [review]

**Pull request merged**